dev/create-release/release-build.sh

```diff
@@ -275,6 +275,8 @@ if [[ "$1" == "package" ]]; then
   # In dry run mode, only build the first one. The keys in BINARY_PKGS_ARGS are used as the
   # list of packages to be built, so it's ok for things to be missing in BINARY_PKGS_EXTRA.
 
+  # NOTE: Don't forget to update the valid combinations of distributions at
+  #   'python/pyspark.install.py' if you're changing them.
   declare -A BINARY_PKGS_ARGS
   BINARY_PKGS_ARGS["hadoop3.2"]="-Phadoop-3.2 $HIVE_PROFILES"
```

Comment:
If we happen to drop Hive 1.2 (or add other combinations of profiles in the distributions), we'll have to change this and [here](https://github.com/apache/spark/pull/29703/files#diff-87e663b6bc59c82beaf09ead1840ac4aR26-R41). I believe this could be done separately later.